lndclient/versioner_client.go

```diff
@@ -0,0 +1,41 @@
+package lndclient
+
+import (
+	"context"
+
+	"github.com/lightningnetwork/lnd/lnrpc/verrpc"
+	"google.golang.org/grpc"
+)
+
+// VersionerClient exposes the version of lnd.
+type VersionerClient interface {
+	// GetVersion returns the version and build information of the lnd
+	// daemon.
+	GetVersion(ctx context.Context) (*verrpc.Version, error)
+}
+
+type versionerClient struct {
+	client      verrpc.VersionerClient
+	readonlyMac serializedMacaroon
+}
+
+func newVersionerClient(conn *grpc.ClientConn,
+	readonlyMac serializedMacaroon) *versionerClient {
+
+	return &versionerClient{
+		client:      verrpc.NewVersionerClient(conn),
+		readonlyMac: readonlyMac,
+	}
+}
+
+// GetVersion returns the version and build information of the lnd
+// daemon.
+//
+// NOTE: This method is part of the VersionerClient interface.
+func (v *versionerClient) GetVersion(ctx context.Context) (*verrpc.Version,
+	error) {
+
+	rpcCtx, cancel := context.WithTimeout(ctx, rpcTimeout)
+	defer cancel()
+	return v.client.GetVersion(rpcCtx, &verrpc.VersionRequest{})
```

Comment:
I think that the macaroon needs to be attached here? I am getting

`[ERR] LOOPD: GetVersion error: rpc error: code = Unknown desc = expected 1 macaroon, got 0`

Reply:
Hmm... I seem to only have tested the negative case and missed this. Thanks for testing!

Reply:
I tried out this, but not sure if it is better (non-blocking of course)
```
rpcCtx, cancel := context.WithTimeout(
	v.readonlyMac.WithMacaroonAuth(ctx), rpcTimeout,
)
```